lib/dockerregistry/inventory.go

```diff
@@ -106,6 +106,30 @@ func MakeSyncContext(
 	return sc, nil
 }
 
+// CollectErrors accepts any number of Errors as input which are then
+// stored in the SyncContext's Logs field.
+func (sc *SyncContext) CollectErrors(errors ...Error) {
+	for _, err := range errors {
+		sc.Logs.Errors = append(sc.Logs.Errors, err)
+	}
+}
+
+// LogJSONSummary logs the SyncContext's Logs as a prettified JSON.
+func (sc *SyncContext) LogJSONSummary() {
+	// Generate the prettified JSON string for klog
+	JSONString := "{\n"
+	JSONString += "\tErrors: [\n"
+	for _, err := range sc.Logs.Errors {
+		JSONString += "\t\t{\n" +
+			"\t\t\tError: " + err.Error.Error() + "\n" +
+			"\t\t\tComment: " + err.Context + "\n" +
+			"\t\t}\n"
+	}
+	JSONString += "\t]\n"
+	JSONString += "}"
+	klog.Info(JSONString)
+}
```

Comment:
this is what I was using to prettify the JSON before I learned of the existence of Marshall functions, so will be changed in future